styleguide-app/Examples/TextInput.elm

```diff
@@ -60,89 +60,106 @@ example =
                 [ Control.view UpdateControl state.control
                     |> Html.fromUnstyled
                 , Heading.h3 [] [ text "TextInput.view { type_ = TextInput.text }" ]
-                , TextInput.view
-                    { label = exampleConfig.label ++ " (text)"
-                    , isInError = exampleConfig.isInError
-                    , placeholder = exampleConfig.placeholder
-                    , showLabel = exampleConfig.showLabel
-                    , value = Maybe.withDefault "" <| Dict.get 1 state.stringInputValues
-                    , onInput = SetTextInput 1
-                    , onBlur = Nothing
-                    , autofocus = False
-                    , type_ = TextInput.text
-                    }
+                , TextInput.view (exampleConfig.label ++ " (text)")
+                    (TextInput.text (SetTextInput 1))
+                    ([ TextInput.errorIf exampleConfig.isInError
+                     , TextInput.placeholder exampleConfig.placeholder
+                     ]
+                        ++ (if exampleConfig.showLabel then
+                                []
+
+                            else
+                                [ TextInput.hiddenLabel ]
```

Comment:
💭 I wonder if `hiddenLabel : Bool -> Attribute msg` might make this a bit nicer?

Alternatively, maybe adding `custom : List (Html.Attribute msg) -> Attribute msg` might help us avoid some parens and brackets

Reply:
Yeah, I was thinking about this...   I think this is only weird in the style-guide.   In real code, each TextInput you create will either have a hidden label or not, it's not going to dynamically change.   I think this is a general issue with elm-debug-controls, in that it makes your "example" code way more dynamic that what you'd actually write when using the component.

I was thinking of just leaving this here for now, and thinking about this a bit longer.  But another option I thought of would be to add a `TextInput.hiddenLabelIf` which could be used here in the styleguide.

Reply:
Or perhaps we could add a `visibleLabel` attribute? In `Button`, we did expose helpers for the defaults even though they're defaults, and I don't think it's been bad.

I don't have a strong opinion about strategy, but I think it is nice if the style guide code is as easy-to-read as possible since people will probably look at it to get code examples.

Reply:
Ah, I realized the debug controls were out of date for the new API, so I changed those to look like this:

<img width="281" alt="Screen Shot 2020-04-14 at 8 55 52 AM" src="https://user-images.githubusercontent.com/1222/79246127-e7742c80-7e2d-11ea-9a6c-123c39398248.png">
<img width="300" alt="Screen Shot 2020-04-14 at 8 56 03 AM" src="https://user-images.githubusercontent.com/1222/79246132-e9d68680-7e2d-11ea-9ca2-73e5e845a7a6.png">

And the styleguide code is now like this:
```
                , TextInput.view (exampleConfig.label ++ " (writing)")
                    (TextInput.text (SetTextInput 4))
                    (List.filterMap identity
                        [ Just TextInput.writing
                        , exampleConfig.maybeErrorAttribute
                        , exampleConfig.maybePlaceholderAttribute
                        , exampleConfig.maybeShowLabelAttribute
                        ]
                    )
                    (Maybe.withDefault "" <| Dict.get 4 state.stringInputValues)
```

which imo is acceptable.  what do you think?

Reply:
Def acceptable! I think I'd vaguely prefer for the error and hidden APIs to match each other a little more closely?

Current version:
```
placeholder : String -> Attribute value msg
onBlur : msg -> Attribute value msg
autofocus : Attribute value msg
hiddenLabel : Attribute value msg
errorIf : Bool -> Attribute value msg
```

Suggested version:
```
placeholder : String -> Attribute value msg
onBlur : msg -> Attribute value msg
autofocus : Bool ->Attribute value msg
hiddenLabelIf : Bool -> Attribute value msg
errorIf : Bool -> Attribute value msg
```

And then rather than a dropdown, we could use the checkbox bool setting from Debug Controls for `hiddenLabelIf`, `autofocus`, and `errorIf`?

Reply:
I made another attempt at cleaning up the use of debug controls here.   It's still not the best, but it gave me an idea of what to add to elm-debug-controls itself to have better support for an attributes-style API.

I do agree the style guide code doesn't look great yet, but I'm also hesitant to add stuff to the public API prematurely just to support the style guide.  I think the API overall is stable enough that any of the additional functions you're suggesting could be added as a minor version bump if they turn out to be useful in real code.

I need to switch over to integrating this into the monolith so we can try to finish a story for our release today, so I'm gonna merge this now.  But I think I understand your concerns, and will continue to think about (and worst case, if we do need a V7, I'll volunteer to write the upgrade script to eliminate V6).